BinaryVault/contracts/BinaryVault.sol

```diff
@@ -0,0 +1,1580 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+import "./interfaces/IBinaryVault.sol";
+import "./interfaces/IBinaryConfig.sol";
+import "./interfaces/ERC721A__IERC721Receiver.sol";
+
+/**
+ * @title Vault of Binary Option Trading
+ * @notice This vault is holding one underlying tokens in it
+ * @author Bunzz Dev, @light-fury
+ */
+contract BinaryVault is
```

Comment:
Please order functions layout by type
Refer https://docs.soliditylang.org/en/v0.8.17/style-guide.html#order-of-layout

Recommendation Order:
State Variables
Events
Modifiers
Constructor
External functions
External functions that are view
External functions that are pure
Public functions
Internal functions
Private functions